### PR TITLE
parser builder unit test

### PR DESCRIPTION
**Pull request by matthewphilyaw**
I think this test case generalizes [parse_test.exs](https://github.com/voluspa/yggdrasil/blob/commands/test/lib/yggdrasil/command/parser_test.exs) and we don't need that test.

@copenhas  your test case descriptions led me in this direction, because really we just want to prove that the parse/1 function parses a given command string into a the struct that the parse defines for that command and sets the field the placeholder specified.

I don't think we need to test Command.Parser directly either, because every test case would like like the on below, I could be wrong and be curious if you can spot a case where that doesn't hold. 

For example if we add an Action and Say command to the Command.Parser module via parse/2 like `parse "say :msg", SayCommand` all we would be testing is that `Command.Parser.parse "say hello"` produces a struct with a field set. If it works for one it works for all and if it fails for one it fails for all. I'm fairly confident in that, but if you can think of a case where it doesn't I'd be curious. 

The edge case where I was thinking it may matter would be if we went into the Move or Say command for example and changed the field, but it actually doesn't compile if you do that. So by it compiling we know that the Struct specified in the parse/2 macro and the field in the placeholder are correct and do exist. 

I feel like if we added 10 commands we end up with lots of duplicated tests that add no value if we were testing the parser directly like parse_test.exs does. Coverage would drop for each command added, but I think that can be made up in other tests like in the player_channel where we would use the Command.Parser to produce the command. Certainly not suggesting we don't test command at all, but rather not sure we need to test the parser directly is all.

### CODE REVIEW

**Comment by copenhas** [comment]
Nice man, really appreciate the test contribution. Are you suggesting we should remove the current parser tests? Or never expand them since there is little value?

**Comment by matthewphilyaw** [comment]
I believe we can remove the current ones and substitute in this one. The  current one uses the MoveCommand explicitly and the intent here is that we consolidate it down to TestCommand/TestParser in one file and test the behavior and leave it at that it's isolated and if we renamed MoveCommand or what have you it doesn't effect this test.

Thoughts?

I feel like expanding them out to cover like MoveCommand and future commands adds little value and like I said it was your test's descriptions that led me down this train of thought when you wrote `"parse/1 sets the fields on the struct based on the placeholder"` - you summed it perfectly there and I was like that is the essence of the test right there.

**Comment by copenhas** [comment]
Sounds good to me. Make the changes in this branch. I'll should be able to review everything tonight.
- Sean Copenhaver

File: test/lib/yggdrasil/command/parser_builder_test.exs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  defstruct foo: nil, bar: nil, baz: nil
+end
+
+defmodule TestParser do
+  use Yggdrasil.Command.ParserBuilder
+
+  parse "test :foo", TestCommand
+  parse "test :foo :bar", TestCommand
+  parse "test :foo :bar :baz", TestCommand
+  parse "test :foo if :bar this :baz matters", TestCommand
```
Comment (matthewphilyaw) [inline test/lib/yggdrasil/command/parser_builder_test.exs:11]: @copenhas I assume this was your intent, and had no idea that was possible till I studied the ParseRule a bit closer and started testing it.

All I have to say brilliant, works well. I had wondered if we were going to support this sort of thing, cause the move commands were simple and it was the only use of it I saw.

**Comment by matthewphilyaw** [comment]
@copenhas last commit sums up what I was aiming for feel free to augment if it need be.

**Comment by copenhas** [comment]
Cool man. I'm going to merge it in. The `commands` branch will get squashed anyway.

**Pull request merged**